service/history/workflow/cache/cache.go

```diff
@@ -147,19 +157,30 @@ func newCache(
 	size int,
 	ttl time.Duration,
 	nonUserContextLockTimeout time.Duration,
+	logger log.Logger,
 	handler metrics.Handler,
 ) Cache {
-	opts := &cache.Options{}
-	opts.TTL = ttl
-	opts.Pin = true
+	opts := &cache.Options{
+		TTL: ttl,
+		Pin: true,
+		OnPut: func(val any) {
+			// TODO: will be used in follow-up PR
+		},
+		OnEvict: func(val any) {
+			// TODO: will be used in follow-up PR
+		},
```

Comment:
This is where the callback has access to (1) the finalizer, (2) the workflow context and (3) the logger.